stdlib/test/collections/test_string.mojo

```diff
@@ -235,60 +235,6 @@ def test_string_join():
     assert_equal(s6, "1,2,3")
 
 
-def test_string_literal_join():
```

Comment:
**Question** Why remove this (unrelated) test?

Reply:
That test is already in `/test/builtin/test_string_literal.mojo` line 222